src/client/components/pages/parts/delete-or-remove-collaboration-modal.js

```diff
@@ -0,0 +1,128 @@
+import * as bootstrap from 'react-bootstrap';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import PropTypes from 'prop-types';
+import React from 'react';
+import request from 'superagent';
+
+
+const {Alert, Button, Modal} = bootstrap;
+
+class DeleteOrRemoveCollaborationModal extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			error: null
+		};
+
+		this.handleSubmit = this.handleSubmit.bind(this);
+	}
+
+	handleSubmit() {
+		request.post(this.postUrl)
+			.send(this.postData)
+			.then((res) => {
+				if (this.props.isDelete) {
+					window.location.href = `/editor/${this.props.collection.ownerId}/collections`;
+				}
+				else {
+					window.location.href = `/editor/${this.props.userId}/collections`;
+				}
+			}, (error) => {
+				this.setState({
+					error: 'Something went wrong! Please try again later'
+				});
+			});
+	}
+
+	render() {
+		const {collection} = this.props;
+		// eslint-disable-next-line one-var
+		let modalBody, modalTitle, submitButton;
+		if (this.props.isDelete) {
+			this.postUrl = `/collection/${collection.id}/delete/handler`;
+			this.postData = {};
+			modalTitle = 'Confirm deletion';
+			modalBody = (
+				<Alert bsStyle="warning">
```

Comment:
```suggestion
				<Alert bsStyle="danger">
```

Let's go even bolder for an irreversible action.